File: fixtures/scale/javascript/large/src/ciphers/nibbleswap.js
// Self-inverse codec (nibbleswap): reverse chunks of 3.
function nibbleswapEncode(text) {
  const k = 3;
  const parts = [];
  for (let i = 0; i < text.length; i += k) {
    parts.push(text.slice(i, i + k).split('').reverse().join(''));
  }
  return parts.join('');
}

function nibbleswapDecode(text) {
  return nibbleswapEncode(text);
}

export default { nibbleswapEncode, nibbleswapDecode };

File: fixtures/scale/lua/large/ciphers/nibbleswap.lua
local function nibbleswap_encode(text)
  local k = 3
  local parts = {}
  for i = 1, #text, k do
    local chunk = text:sub(i, i + k - 1)
    parts[#parts + 1] = chunk:reverse()
  end
  return table.concat(parts)
end

local function nibbleswap_decode(text)
  return nibbleswap_encode(text)
end

return { nibbleswap_encode = nibbleswap_encode, nibbleswap_decode = nibbleswap_decode }
